utils/build-script

```diff
@@ -1473,6 +1514,191 @@ details of the setups of other systems or automated environments.""")
+            # SDK? Is that right?
+            if not should_build_stdlib_target(stdlib_target):
+                continue
+
+            # Compute build and test information.
+            build = stdlib_target.platform not in platforms_to_skip_build
+            test = stdlib_target.platform not in platforms_to_skip_test
+            test_host_only = None
+            build_benchmark = build and stdlib_target.supports_benchmark
+            # FIXME: Note, `build-script-impl` computed a property here
+            # w.r.t. testing, but it was actually unused.
+
+            # As a special case, Darwin embedded non-simulator platforms have
+            # different behavior for the test condition.
+            if stdlib_target.platform.is_darwin and \
+               stdlib_target.platform.is_embedded and \
+               not stdlib_target.platform.is_simulator:
+                if stdlib_target.platform not in platforms_to_skip_test_host:
+                    test_host_only = True
+                    test = True
+                else:
+                    test = False
+
+            name = stdlib_target.name
+            if build:
+                # FIXME: This logic doesn't make much sense, but
+                # build-script-impl had no comments.
+                if args.build_swift_stdlib_unittest_extra or \
+                       args.validation_test or args.long_test:
+                    swift_stdlib_targets.append("swift-stdlib-"+name)
+                else:
+                    swift_stdlib_targets.append("swift-test-stdlib-"+name)
+            if build_benchmark:
+                swift_benchmark_targets.append("swift-benchmark-"+name)
+                # FIXME: This probably should respect `args.benchmark`, but a
+                # typo (?) in build-script-impl meant we always would do this.
+                swift_run_benchmark_targets.append(
+                    "check-swift-benchmark-"+name)
+            if test:
+                if test_host_only:
+                    suffix = "-non-executable"
+                else:
+                    suffix = ""
+                subset_suffix = ""
+                if args.validation_test and args.long_test:
+                    subset_suffix = "-all"
+                elif args.validation_test:
+                    subset_suffix = "-validation"
+                elif args.long_test:
+                    subset_suffix = "-only_long"
+                else:
+                    subset_suffix = ""
+                swift_test_targets.append("check-swift{}{}-{}".format(
+                    subset_suffix, suffix, name))
+                if args.test_optimized and not test_host_only:
+                    swift_test_targets.append(
+                        "check-swift{}-optimize-{}".format(
+                            subset_suffix, name))
+
+        return {
+            "SWIFT_STDLIB_TARGETS": " ".join(swift_stdlib_targets),
+            "SWIFT_SDKS": " ".join(sorted(swift_sdks)),
```

Comment:
Yeah, I understand. The reason why we sort here is so that the refactor validation logic inside `build-script-impl` works (because we compute the exact same result).